File: Analysis/python/plotpostfit.py
from task import Task

import os

class PlotPostfit(Task):
    def __init__(self, config = {}):
        super().__init__(config)

    def run(self):
        self["executable"] = "PlotPostfit"

        self["arguments"] = [
                "--limit-dir", self["limit-dir"],
                "--mass", self["mass"],
                "--channels", *self["channels"],
                "--out-dirs", self["dir"],
        ]

    def output(self):
        self["output"] = self["dir"] + "/postfit_{}.pdf".format(self["mass"])

    @staticmethod
    def configure(config, mass):
        tasks = []

        limitConf = {
                    "name": "Postfitplot_{}".format(mass),
                    "limit-dir": "{}/{}".format(os.environ["CHDIR"], config["dir"]), 
                    "dir":  "{}/{}".format(os.environ["CHDIR"], config["dir"]), 
                    "display-name": "Postfit: {}".format(mass), 
                    "dependencies": ["Limit_{}".format(mass)],
                    "channels": config["channels"],
                    "mass": mass,
        }

        tasks.append(PlotPostfit(limitConf))

        return tasks

File: Analysis/bin/plotpostfit.cc
#include <string>
#include <vector>

#include <ChargedAnalysis/Utility/include/parser.h>
#include <ChargedAnalysis/Analysis/include/plotterPostfit.h>

int main(int argc, char* argv[]){
    //Parser arguments
    Parser parser(argc, argv);

    int mass = parser.GetValue<int>("mass");
    std::string limitDir = parser.GetValue<std::string>("limit-dir");
    std::vector<std::string> channels = parser.GetVector<std::string>("channels");
    std::vector<std::string> outDirs = parser.GetVector<std::string>("out-dirs");
  
    //Call and run Plotter1D class
    PlotterPostfit plotter(limitDir, mass, channels);
    plotter.ConfigureHists();
    plotter.Draw(outDirs);
}
